packages/driver/src/cypress/cy.js

```diff
@@ -713,7 +713,8 @@ const create = function (specWindow, Cypress, Cookies, state, config, log) {
     // attached to the error thrown from chai
     // command errors and command assertion errors (default assertion or cy.should)
     // have the invocation stack attached to the current command
-    let userInvocationStack = state('currentAssertionUserInvocationStack')
+    // prefer err.userInvocation stack if it's been set
+    let userInvocationStack = $errUtils.getUserInvocationStack(err) || state('currentAssertionUserInvocationStack')
```

Comment:
Why `$errUtils.getUserInvocationStack(err)` and not just `err.userInvocationStack`?

Reply:
brian's reasoning was we dont want magical properties on the error instance without going through an explicit contract